osu.Game.Rulesets.Taiko/UI/DrumTouchInputArea.cs

```diff
@@ -0,0 +1,141 @@
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Framework.Logging;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Taiko.UI
+{
+    /// <summary>
+    /// An overlay that captures and displays Taiko mouse and touch input.
+    /// The boundaries of this overlay defines the interactable area for touch input.
+    /// A secondary InputDrum is attached by this overlay, which defines the circulary boundary which distinguishes "centre" from "rim" hits, and also displays input.
+    /// </summary>
+    public class DrumTouchInputArea : Container
+    {
+        // The percent of the drum that extends past the bottom of the screen (set to 0.0f to show the full drum)
+        private const float overhangPercent = 0.33f;
+        private readonly InputDrum touchInputDrum;
+
+        [Resolved(canBeNull: true)]
+        private TaikoInputManager taikoInputManager { get; set; }
+
+        private KeyBindingContainer<TaikoAction> keyBindingContainer;
+
+        // Which Taiko action was pressed by the last OnMouseDown event, so that the corresponding action can be released OnMouseUp even if the cursor position moved
+        private TaikoAction mouseAction;
+
+        // A map of (Finger Index OnTouchDown -> Which Taiko action was pressed), so that the corresponding action can be released OnTouchUp is released even if the touch position moved
+        private Dictionary<TouchSource, TaikoAction> touchActions = new Dictionary<TouchSource, TaikoAction>(Enum.GetNames(typeof(TouchSource)).Length);
+
+        private Playfield playfield;
+
+        public DrumTouchInputArea(Playfield playfield) {
+            this.playfield = playfield;
+
+            RelativeSizeAxes = Axes.Both;
+            RelativePositionAxes = Axes.Both;
+            Children = new Drawable[]
+            {
+                new Box() {
+                    Alpha = 0.0f,
+                    Colour = new OsuColour().Blue,
+
+                    RelativeSizeAxes = Axes.Both,
+                    RelativePositionAxes = Axes.Both,
+                },
+                new Container() {
+                    RelativeSizeAxes = Axes.Both,
+                    RelativePositionAxes = Axes.Both,
+
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.BottomCentre,
+
+                    Children = new Drawable[]
+                    {
+                        touchInputDrum = new InputDrum(playfield.HitObjectContainer) {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                        },
+                    }
```

Comment:
Not sure why drawables are being used here if they aren't shown. I'd expect the overlay to appear in some way (showing the borders at least) on first mouse/touch input that is recognised, or eventually if a device is detected as a touch device.

Reply:
The box is intended to be removed after I'm done testing this.

On the note you mentioned about the overlay only appearing when touch input is detected - I agree. I'll try and get to that next once I got the basic implementation down.